lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -5553,6 +5553,51 @@ class DecomposeAtenInstanceNormOp
 };
 } // namespace
 
+namespace {
+class DecomposeAten_WeightNormInterfaceOp
+    : public OpRewritePattern<Aten_WeightNormInterfaceOp> {
+    using OpRewritePattern<Aten_WeightNormInterfaceOp>::OpRewritePattern;
+    LogicalResult matchAndRewrite(Aten_WeightNormInterfaceOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Location loc = op.getLoc();
+    Value v = op.getV();
+    Value g = op.getG();
+    Value dim = op.getDim();
+
+    auto inputType = cast<BaseTensorType>(v.getType());
+    if (!inputType.hasSizes())
+      return rewriter.notifyMatchFailure(op, "expected input to have sizes");
+
+    auto sizes = inputType.getSizes();
+    SmallVector<Value> keepDims;
+    for (size_t i = 0; i < sizes.size(); ++i) {
+      if (i != dim.getDefiningOp<ConstantIntOp>().getValue())
```

Comment:
You will need to check that `dim` has a `ConstantIntOp` as its defining operation. This will return `null` if its not.

Reply:
Fixed it in the latest commit